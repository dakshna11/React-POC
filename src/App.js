import './App.css';
import './Components/styles.css'
import Footer from './Components/Footer';
import Header from './Components/Header';
import TopBar from './Components/TopBar';
import HeaderSub from './Components/Header-comp/HeaderSub';

function App() {
  return (
    <div>
      <TopBar/>
      <Header/>
      <HeaderSub/>
      <h1>Body contents</h1>
      <Footer/>
    </div>
  );
}

export default App;
