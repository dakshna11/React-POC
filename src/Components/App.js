
import '../Components/styles/styles.css'
import {BrowserRouter as Router , Route ,Switch } from 'react-router-dom'
import Footer from './Footer';
import Header from './Header';
import TopBar from './TopBar';
import Home from './Header-comp/Home';
import WashingPoints from './Header-comp/WashingPoints';
import About from './Header-comp/About';
import Contact from './Header-comp/Contact';
import DemoCarousel from './Header-comp/Carousel';
import Price from './Header-comp/Price';
import Service from './Header-comp/Service';


function App() {
  window.onscroll = () => {scrollFunction()};

const scrollFunction=()=> {
  if  (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20) {
    document.getElementById("mybtn").style.display = "block";
  } else {
    document.getElementById("mybtn").style.display = "none";
  }
}


const topFunction= () => {
  document.body.scrollTop = 0;
  document.documentElement.scrollTop = 0;
}


  return (
    <div>
      <Router>
      <TopBar/>
      <Header/>
      <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/about' component={About}/>
            <Route path='/Price' component={Price}/>
            <Route path='/Washing Points' component={WashingPoints}/>
            <Route path='/Contact' component={Contact}/>
            <Route path='/Service' component={Service}/>
      </Switch>
      <Footer/>
      <button onClick={topFunction} id="mybtn"><i className='fa fa-chevron-up'></i></button>
      </Router>
    </div>
  );
}
export default App;
