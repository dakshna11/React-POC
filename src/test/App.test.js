import React from 'react';
import { shallow } from 'enzyme';
import App from "../Components/App";


let wrapped = shallow(<App></App>);
describe('App', () => {
  it('should render the App Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
  // it('test the function' , () => {
  //   expect(wrapped.document.body.scrollTop).toBe('0')

  // })
})