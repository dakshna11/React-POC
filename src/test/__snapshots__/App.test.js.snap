// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App Component correctly 1`] = `
<div>
  <BrowserRouter>
    <TopBar />
    <Header />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/Price"
      />
      <Route
        component={[Function]}
        path="/Washing Points"
      />
      <Route
        component={[Function]}
        path="/Contact"
      />
      <Route
        component={[Function]}
        path="/Service"
      />
    </Switch>
    <Footer />
    <button
      id="mybtn"
      onClick={[Function]}
    >
      <i
        className="fa fa-chevron-up"
      />
    </button>
  </BrowserRouter>
</div>
`;
